### ISSUE TITLE
RabbitMQ failing after install due to Erlang version 

### ISSUE THREAD

**Issue by tonyppe** [cmacrae/ansible-sensu#180]
I decided to re-deploy sensu_masters on a fresh ubuntu 16 instance and the playbook fails during the part where it checks the vhost on the rabbitmq server. I figured out that the reason for the failure is due to the erlang version being 21 where the rabbitmq doc states it must be fixed to 20.1 (ref: https://www.rabbitmq.com/install-debian.html) 

After seeing the playbook error I went to the instance and run this command and had the below output. Note, it's not prod instance, I am trying to work forwards as though it is. 
```
root@sensuprod1:/home/ansible# sudo rabbitmqctl status
sudo: unable to resolve host sensuprod1: Connection timed out
=SUPERVISOR REPORT==== 22-Jun-2018::04:49:28.812755 ===
    supervisor: {local,'Elixir.Logger.Supervisor'}
    errorContext: start_error
    reason: noproc
    offender: [{pid,undefined},
               {id,'Elixir.Logger.ErrorHandler'},
               {mfargs,
                   {'Elixir.Logger.Watcher',start_link,
                       [{error_logger,'Elixir.Logger.ErrorHandler',
                            {true,false,500}}]}},
               {restart_type,permanent},
               {shutdown,5000},
               {child_type,worker}]
=CRASH REPORT==== 22-Jun-2018::04:49:28.830907 ===
  crasher:
    initial call: application_master:init/4
    pid: <0.81.0>
    registered_name: []
    exception exit: {{shutdown,
                         {failed_to_start_child,'Elixir.Logger.ErrorHandler',
                             noproc}},
                     {'Elixir.Logger.App',start,[normal,[]]}}
      in function  application_master:init/4 (application_master.erl, line 138)
    ancestors: [<0.80.0>]
    message_queue_len: 1
    messages: [{'EXIT',<0.82.0>,normal}]
    links: [<0.80.0>,<0.42.0>]
    dictionary: []
    trap_exit: true
    status: running
    heap_size: 376
    stack_size: 27
    reductions: 193
  neighbours:

=CRASH REPORT==== 22-Jun-2018::04:49:28.812687 ===
  crasher:
    initial call: Elixir.Logger.Watcher:init/1
    pid: <0.88.0>
    registered_name: []
    exception exit: noproc
      in function  gen:do_for_proc/2 (gen.erl, line 228)
      in call from gen_event:rpc/2 (gen_event.erl, line 239)
      in call from 'Elixir.Logger.Watcher':init/1 (lib/logger/watcher.ex, line 23)
      in call from gen_server:init_it/2 (gen_server.erl, line 374)
      in call from gen_server:init_it/6 (gen_server.erl, line 342)
    ancestors: ['Elixir.Logger.Supervisor',<0.82.0>]
    message_queue_len: 0
    messages: []
    links: [<0.83.0>]
    dictionary: []
    trap_exit: true
    status: running
    heap_size: 376
    stack_size: 27
    reductions: 254
  neighbours:

=INFO REPORT==== 22-Jun-2018::04:49:28.835332 ===
    application: logger
    exited: {{shutdown,
                 {failed_to_start_child,'Elixir.Logger.ErrorHandler',noproc}},
             {'Elixir.Logger.App',start,[normal,[]]}}
    type: temporary
Could not start application logger: Logger.App.start(:normal, []) returned an error: shutdown: failed to start child: Logger.ErrorHandler
    ** (EXIT) no process: the process is not alive or there's no process currently associated with the given name, possibly because its application isn't started
```

To work around the problem I done the following:

1. purged erlang with 'sudo apt-get purge erlang*'
2. downloaded and installed erlang 20.1 from this guide (only step 2): https://www.vultr.com/docs/how-to-install-rabbitmq-on-ubuntu-16-04-47
3. re-installed rabbitmq-server (note apt was complaining about some missing packages at this point and requested that I run 'apt-get -f install'

After rabbitmq was reinstalled, I checked the failing command is now successful (ie not crashing):
```
ansible@sensuprod1:~$ sudo /usr/sbin/rabbitmqctl -q -n rabbit list_vhosts name tracing
sudo: unable to resolve host sensuprod1: Connection timed out
/       false
/sensu  false
```
After that, the playbook was still failing so I rebooted the instance which seemed to have resolved that failure (shrug shoulders).

Rabbitmq guide :: https://www.rabbitmq.com/install-debian.html#apt-pinning

**Comment by jaredledvina** [comment]
Hi @tonyppe 

So, looking at https://github.com/sensu/sensu-ansible/blob/master/tasks/Ubuntu/rabbit.yml,  we already make use of their Bintray mirrors for the latest RabbitMQ & RabbitMQ-Erlang packages. 

However, when we install `rabbitmq-server` it looks like we don't also list out erlang: https://github.com/sensu/sensu-ansible/blob/master/tasks/Ubuntu/rabbit.yml#L33
My idea being that `rabbitmq-server` doesn't properly have the dependencies listed out. Now, that said, RabbitMQ doesn't _require_ Erlang 20.X ~~and does work with 21.X (since that's what I'm running in production)~~ . It's possible that there's some weirdness with the Ubuntu 16.04 specific set though. 

I would be very useful if you know of there were any error messages during the role deployment, which version of Ansible you're using, and which version of this role you're deploying.

**Comment by frankjoshua** [comment]
ansible 2.5.5
master version of this role

Same issue for me. I use vagrant and virtualbox for testing and it's failing with the same error with this config.
```
  config.vm.define "server2" do |server2|
    server2.vm.network :private_network, ip: "192.168.33.57"
    #port for uchiwa
    server2.vm.network "forwarded_port", guest: 3000, host: 3000
    #Base box with ubuntu 16.04 
    server2.vm.box = "ubuntu/xenial64"
    server2.vm.provider "virtualbox" do |vb|
      # Customize the amount of memory on the VM:
      vb.memory = "1024"
      vb.cpus = 2
    end
  end
```

**Comment by frankjoshua** [comment]
Here is a work around for now. Add this pre task to make sure version 20 of Erlang is used since RabbitMQ does not work with version 21.
 ```yaml
pre_tasks:
  - name: Make sure the correct version of Erlang is used
    copy:
      dest: "/etc/apt/preferences.d/erlang"
      content: |
          Package: esl-erlang erlang*
          Pin: version 1:20*
          Pin-Priority: 999
    become: yes
```

**Comment by jaredledvina** [comment]
@frankjoshua - Thanks for the workaround! It also looks like the automated integration tests have caught this as well: https://travis-ci.org/sensu/sensu-ansible/jobs/396910827#L979 

I'll try to get some time this weekend to PR a fix and cut another release. If anyone wants to toss a PR my way as well, just let me know. 

As for the scope of the issue, it seems like it's actually affect every OS except CentOS and AmazonLinux: https://travis-ci.org/sensu/sensu-ansible/builds/396910825

**Comment by jaredledvina** [comment]
Oh maaan, there's some cruft we need to clean up in this role regarding Erlang:
Amazonlinux:
* Installs the same GPG key, twice: https://github.com/sensu/sensu-ansible/blob/master/tasks/Amazon/rabbit.yml#L11-L13
* Is accidentally pinned to erlang 19: https://github.com/sensu/sensu-ansible/blob/master/tasks/Amazon/rabbit.yml#L29

CentOS:
* Installs the same GPG key twice: https://github.com/sensu/sensu-ansible/blob/master/tasks/CentOS/rabbit.yml#L11-L13
* Is accidentally pinned to erlang 19: https://github.com/sensu/sensu-ansible/blob/master/tasks/CentOS/rabbit.yml#L29

Debian:
* Installs erlang from erlang-solutions not the rabbitmq's erlang repo's which might be because they don't publish `.deb` packages but, is why it's getting a newer version: https://github.com/sensu/sensu-ansible/blob/master/tasks/Debian/rabbit.yml#L19-L29

Fedora:
* Missing a lot of the GPG configs: https://github.com/sensu/sensu-ansible/blob/master/tasks/Fedora/rabbit.yml versus https://github.com/sensu/sensu-ansible/blob/master/tasks/CentOS/rabbit.yml#L7-L14 (this might actually be fine, I'm forgetting how Fedora does it's GPG setup off hand)
* Accidentally is pinned to Erlang 19: https://github.com/sensu/sensu-ansible/blob/master/tasks/Fedora/rabbit.yml#L20

Ubuntu:
* Like Debian, also uses erlang-solutions: https://github.com/sensu/sensu-ansible/blob/master/tasks/Ubuntu/rabbit.yml#L19-L29 and https://github.com/sensu/sensu-ansible/blob/master/vars/Ubuntu.yml#L10

So, there's a bit of cleanup but, the primary issue is that Debian based OS's are currently using `erlang-solutions` for packages which have Erlang 21 available versus the RabbitMQ Erlang repo's which currently only have Erlang 20 published as RPM's. Ideally, RabbitMQ also publishes Erlang `.deb` packages and we can switch all of the OS's to a single repo here but, I'll have to look further into that. 

If anyone want's to PR fixes for the above, feel free to give it a whirl. I'll work on this as time allows as well.

**Comment by jaredledvina** [comment]
So yeah, I _think_ what makes sense is to re-work the RabbitMQ/Erlang install setup across all of the OS's such that we ensure everything follows the current recommendations on http://www.rabbitmq.com/which-erlang.html 

Both http://www.rabbitmq.com/install-rpm.html and http://www.rabbitmq.com/install-debian.html recommend locking the erlang version. I believe I can actually have every OS using Bintray for RabbitMQ and Erlang Solutions for erlang and then Pin Erlang and RabbitMQ via some variables. Doing that should resolve this bug while allow everyone to pin to whichever version they are most comfortable with.

**Comment by jaredledvina** [comment]
PR merged and fix releases in 2.6.0!
https://github.com/sensu/sensu-ansible/releases/tag/2.6.0
